pkg/polylog/polyzero/logger_test.go

```diff
@@ -0,0 +1,369 @@
+		},
+		{
+			Key:                    "Float64",
+			Value:                  float64(420.69),
+			ExpectedOutputContains: `"Float64":420.69`,
+		},
+		{
+			EventMethodName:        "Err",
+			Value:                  expectedErr,
+			ExpectedOutputContains: `"error":"42"`,
+		},
+		{
+			EventMethodName:        "Timestamp",
+			ExpectedOutputContains: expectedTimestampEventContains,
+		},
+		// TODO_TECHDEBT: figure out why this fails in CI but not locally,
+		// (even with `make itest 500 10 ./pkg/polylog/... -- -run=ZeroLogger_AllLevels_AllEventTypeMethods`).
+		//
+		//{
+		//	Key:                    "Time",
+		//	Value:                  expectedTime,
+		//	ExpectedOutputContains: expectedTimeEventContains,
+		//},
+		{
+			Key:                    "Dur",
+			Value:                  expectedDuration,
+			ExpectedOutputContains: expectedDurationEventContains,
+		},
+		{
+			EventMethodName: "Fields",
+			Value: map[string]any{
+				"key1": "value1",
+				"key2": 42,
+			},
+			ExpectedOutputContains: `"key1":"value1","key2":42`,
+		},
+		{
+			EventMethodName:        "Fields",
+			Value:                  []any{"key1", "value1", "key2", 42},
+			ExpectedOutputContains: `"key1":"value1","key2":42`,
+		},
+	}
+
+	for _, level := range polyzero.Levels() {
+		testpolylog.RunEventMethodTests(
+			t,
+			level,
+			tests,
+			newTestLogger,
+			newTestEventWithLevel,
+			"*polyzero.zerologEvent",
+			getExpectedLevelOutputContains,
+		)
+	}
+}
+
+func TestZerologLogger_Levels_Discard(t *testing.T) {
+	// Construct a logger with each level. With each logger, log an event at each
+	// level and assert that the event is logged if and only if the event level
+	// is GTE the logger level.
+	for _, loggerLevel := range polyzero.Levels() {
+		testDesc := fmt.Sprintf("%s level logger", loggerLevel.String())
+		t.Run(testDesc, func(t *testing.T) {
+			logger, logOutput := newTestLogger(t, loggerLevel)
+
+			// Log an event for each level.
+			for _, eventLevel := range polyzero.Levels() {
+				event := newTestEventWithLevel(t, logger, eventLevel)
+				// Log the event level string.
+				event.Msg(eventLevel.String())
+
+				// If the event level is GTE the logger level, then the event should
+				// be logged.
+				if eventLevel.Int() >= loggerLevel.Int() {
+					require.Truef(t, event.Enabled(), "expected event to be enabled")
+					require.Contains(t, logOutput.String(), eventLevel.String())
+				} else {
+					require.Falsef(t, event.Enabled(), "expected event to be discarded")
+					require.NotContains(t, logOutput.String(), eventLevel.String())
+				}
+			}
+
+			// Print log output for manual inspection.
+			t.Log(logOutput.String())
+		})
+	}
+}
+
+func TestZerologLogger_Func_Discard_Enabled(t *testing.T) {
+	for _, loggerLevel := range polyzero.Levels() {
+		testDesc := fmt.Sprintf("%s loggerLevel logger", loggerLevel.String())
+		t.Run(testDesc, func(t *testing.T) {
+			var (
+				notExpectedOutput = "if you're reading this, the test failed"
```

Comment:
Here I am reading it on github and am now facing an existential crisis.

Reply:
Don't worry, in this scenario you're a god outside of time :wink: 
![image](https://github.com/pokt-network/poktroll/assets/600733/10eb2ca7-2e00-4de2-b970-207b15b9fdfd)